src/statsupdate.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+
+import datetime
+from email.mime.application import MIMEApplication
+
+import config
+import mail
+import donationswap
+
+'''
+xxx do not hardcode paths
+'''
+CONFIG_FILENAME = '/srv/web/app-config.json'
+CONFIG = config.Config(CONFIG_FILENAME)
+
+def send_mail(msg, to, filename, data):
+	m = mail.Mail(CONFIG.email_user, CONFIG.email_password, CONFIG.email_smtp, CONFIG.email_sender_name)
+
+	smtp_msg = m._prepare_msg('Donation Swap Stats Update', msg, msg, to, None, None)
+
+	part = MIMEApplication(data, Name=filename)
+	part['Content-Disposition'] = 'attachment; filename="' + filename + '"'
+	smtp_msg.attach(part)
+
+	m._send_msg(smtp_msg)
+
+def makeData(data):
+	compiledData = "date match generated, USD value, charity1, country1, charity2, country2\n"
+
+	for match in data:
+		compiledData += "{}, {}, {}, {}, {}, {}\n".format(match['created_ts'], match['value'], match['details']['new_offer_charity'], match['details']['new_offer_country'], match['details']['old_offer_charity'], match['details']['old_offer_country'])
```

Comment:
one of these might be more readable
 * `'%(created_ts)s, %(value)s, ...' % match` (doesn't work so well for sub-sub thingies
 * `f"{match['created_ts']} {match['value']}..."` (depends on installed python versions)

Reply:
I believe the % operator for strings isn't intended to continue into the future.
Happy to try and rewrite it cleaner if you'd prefer, but i figured I'd go with something basic that shouldn't complain if something changes

Reply:
(merging, feel free to ping me any time to fix this)